modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/DLManagementToolbarPropsTransformer.js

```diff
@@ -0,0 +1,252 @@
+					`#${portletNamespace}changeLog`
+				);
+
+				if (changeLogInput) {
+					changeLogInput.setAttribute('value', changeLog);
+				}
+
+				const versionIncreaseInput = form.querySelector(
+					`#${portletNamespace}versionIncrease`
+				);
+
+				if (versionIncreaseInput) {
+					versionIncreaseInput.setAttribute('value', versionIncrease);
+				}
+
+				processAction('checkin', editEntryURL);
+			});
+		});
+	};
+
+	const deleteEntries = () => {
+		let action;
+
+		if (trashEnabled) {
+			action = 'move_to_trash';
+		}
+		else if (
+			confirm(
+				Liferay.Language.get(
+					'are-you-sure-you-want-to-delete-the-selected-entries'
+				)
+			)
+		) {
+			action = 'delete';
+		}
+
+		processAction(action, editEntryURL);
+	};
+
+	const editTags = () => {
+		const searchContainer = Liferay.SearchContainer.get(
+			otherProps.searchContainerId
+		);
+
+		Liferay.componentReady(`${portletNamespace}EditTagsComponent`).then(
+			(editTagsComponent) => {
+				const bulkSelection = searchContainer.select?.get(
+					'bulkSelection'
+				);
+
+				const selectedFileEntries = searchContainer.select
+					.getAllSelectedElements()
+					.get('value');
+
+				editTagsComponent.open(
+					selectedFileEntries,
+					bulkSelection,
+					folderConfiguration.defaultParentFolderId
+				);
+			}
+		);
+	};
+
+	const move = () => {
+		const searchContainer = Liferay.SearchContainer.get(
+			otherProps.searchContainerId
+		);
+
+		let selectedItems = 0;
+
+		if (searchContainer.select) {
+			selectedItems = searchContainer.select
+				.getAllSelectedElements()
+				.filter(':enabled')
+				.size();
+		}
+
+		const dialogTitle =
+			selectedItems === 1
+				? Liferay.Language.get('select-destination-folder-for-x-item')
+				: Liferay.Language.get('select-destination-folder-for-x-items');
+
+		openSelectionModal({
+			height: '480px',
+			id: `${portletNamespace}selectFolder`,
+			onSelect(selectedItem) {
+				const newFolderId = selectedItem.folderid;
+
+				const form = document.getElementById(`${portletNamespace}fm2`);
+
+				if (!form) {
+					return;
+				}
+
+				form.setAttribute('action', editEntryURL);
+				form.setAttribute('enctype', 'multipart/form-data');
+				form.setAttribute('method', 'post');
+
+				const cmdInput = form.querySelector(`#${portletNamespace}cmd`);
+
+				if (cmdInput) {
+					cmdInput.setAttribute('value', 'move');
+				}
+
+				const newFolderIdInput = form.querySelector(
+					`#${portletNamespace}newFolderId`
+				);
+
+				if (newFolderIdInput) {
+					newFolderIdInput.setAttribute('value', newFolderId);
+				}
+
+				submitForm(form, editEntryURL, false);
+			},
+			selectEventName: `${portletNamespace}selectFolder`,
+			size: 'lg',
+			title: Liferay.Util.sub(dialogTitle, [selectedItems]),
+			url: selectFolderURL,
+		});
+	};
+
+	return {
+		...otherProps,
+		onActionButtonClick(event, {item}) {
+			const action = item.data?.action;
+
+			if (action === 'checkin') {
+				checkIn();
+			}
+			if (action === 'checkout') {
```

Comment:
shouldn't this be an `else if`?

Anyway, do we prefer this if-else structure better than a `switch`? (Just asking)

Reply:
you're right missing an `else` here, good catch. I'll that right now.

Concerning the `switch`, that's a good question, but we've been using this "pattern" so far so I'm just applying the same here.

Reply:
I updated that in [https://github.com/liferay-lima/liferay-portal/pull/1525/commits/82125cf58a4192f968a5906ff3e2cfd5a02aeb07](https://github.com/liferay-lima/liferay-portal/pull/1525/commits/82125cf58a4192f968a5906ff3e2cfd5a02aeb07), thanks for the review @carloslancha (I'm missing the PR is not horrible comments :joy:)

Reply:
This pr is not horrible at all :P